src/stakingSetup.js

```diff
@@ -0,0 +1,153 @@
+                        throw [APP_FAIL_TRANSACTION_REJECTED, `Transaction rejected ${result.status} for ${account.meta.name}`];
+                    }
+                    cennz();
+                }
+            })
+            .catch(err => transaction_error = err);
+            await sleep();
+            
+    }
+
+    for (account of stashAccount) {
+        log.info("SETTING UP FOR CENTRAPAY -------------- \n Accout raw seed: " + account.meta.name)
+        log.info(`Transfering centrapay to ${account.meta.name} and address ${account.address}`)
+        const unsub = await api.tx.genericAsset.transfer(16001, account.address, "100_000_000_000_000")
+            .signAndSend(acliceKeyring, (result) => {
+                log.info(`Current status is ${result.status}`);
+
+                if (result.isCompleted) {
+                    if (result.isFinalized) {
+                        hash = result.status.asFinalized;
+                    }
+                    else { // error
+                        log.info("Transaction Rejected");
+                        throw [APP_FAIL_TRANSACTION_REJECTED, `Transaction rejected ${result.status}`];
+                    }
+                    unsub();
+                }
+            })
+            .catch(err => transaction_error = err);
+            await sleep();
+    }
+
+    await bondStashAccounts(api, stashAccount);
+    await setSessionKey(api);
+    await setNodesValidating(api);
+}
+
+function sleep(){
+    return new Promise(resolve=>{
+        setTimeout(resolve,DEFAULT_SLEEP)
+    })
+}
+
+async function bondStashAccounts(api, stashes){
+    
+    for(i=0; i< controllerAddrs.length; i++){
+       let transfer = api.tx.staking.bond(controllerAddrs[i], "5_000_000", 0);
+        // Sign and Send the transaction
+        transfer.signAndSend(stashes[i], ({ events = [], status, type }) => {
+            if (type === 'Finalised') {
+                console.log('Successful transfer ' + ' with hash ' + status.asFinalised.toHex());
+              } else {
+                console.log('Status of transfer: ' + type);
+              }
+              events.forEach(({ phase, event: { data, method, section } }) => {
+                console.log(phase.toString() + ' : ' + section + '.' + method + ' ' + data.toString());
+              });
+        })
+        .catch(err => transaction_error = err);
+
+        await sleep();
+    }
+}
+
+async function setSessionKey(api){
+    sessionKeys =[
+        "5Cdw3BqUmFEdnEGQi62vk6E27XsL22R1A51Xea5xqU7xTDCc", //BOB
+        "5Dvtpw1vdHcqZNURRkVP6izMPP1VcpUyUXmHYnGL2pDz5Ef2", //CHARLIE
+        "5DMEX3mnK2vmTX4BfGpQXVoS2ST6PhgCQ56JkGdEXSnzZ3s3" //EVE
+
+    ];
+
+    for(i=0; i<sessionKeys.length; i++){
+        try{
+            const transfer = await api.tx.session.setKey(sessionKeys[i]);
+            transfer.signAndSend(controllersKeyring[i])
+            
+        }catch(err){
+            console.log(err)
+        };
+
+        await sleep();
```

Comment:
race condition - semiphore required here too
> Note, the reason I allow race conditions to occur during load testing is because part of the test spec is to check whether the transaction finalizes in a sensible amount of time.  However, we don't want to do this check during setup